### ISSUE TITLE
Ability to ignore arguments in specific method calls

### ISSUE THREAD

**Issue by cervengoc** [EWSoftware/VSSpellChecker#51]
When somebody (like me now) uses a non-english dictionary and language, then almost every "technical" string (like parameter names in ArgumentException constructor, etc.) in every method call is treated as misspelled word. I could add all of them to the ignore list but that's a huge work.

It would be a very helpful feature if one could somehow add a method to the ignore list, marking that the calls to that method should be never checked.

Without this feature, this great pugin seems to be almost unusable with a non-english language.

**Comment by cervengoc** [comment]
UPDATE: Actually I've just realized the multilanguage checking support, which improved some things, but still very uncomfortable, since I use a lot of technical strings, for example in Razor HTML helpers (CSS class names, several IDs, etc.) which are not correct in any language. I would still like to exclude them based on method calls instead of one-by-one.

**Comment by EWSoftware** [comment]
Are these method signatures in comments? Can you give an example?

**Comment by cervengoc** [comment]
An example could be a method call in code like this:

```
@Html.MyButton("Save", id: "btn-save", classes: "btn btn-default my-btn-class")
```

But such thing can appear anywhere, it's just likely in Razor code for me now.
In the example, the spell checker should check only the label of the button, "Save". The other parameters shoulw ideally be treated as "technical" strings. However, the plugin underlines every part of the dashed strings, etc.

I hope I managed to clarify the situation.

**Comment by EWSoftware** [comment]
Can you provide a full simple example?  A single line doesn't provide enough context to see what you are trying to convey.  Bear in mind that I don't do any web development so Razor code doesn't mean anything to me.  If you can e-mail me a page or small test project that would work best so that I can open it and see what you are seeing.

**Comment by cervengoc** [comment]
It's very simple, Razor was just an edge case. But the issue happens in pure C# too. For example there are underlinings and correction suggestions:
- at dictionary string keys
- format arguments
- other technical strings like WinForms control names, etc.

Also, it's very annoying that in Javascript files, all of my jQuery selectors have some underlined parts which is completely false.

The plugin could be much more usable if it could somehow distinguish between technical strings and content strings, if you know what I mean. I can see that it's a very hard task, that's why I suggest adding some features that users can "teach" the plugin, for example by marking methods like "wherever THIS method gets called, the arguments should not be checked". 

Another helpful way was to add the ability for us to define ignored patterns based on regular expressions. Then I could add for example that every text beginning with "btn..." shouldn't be checked because it's most likely an ID of a button control (or web element).

**Comment by EWSoftware** [comment]
I doubt what you are asking for is possible since you can't really know what the content of a string is especially if it's just appearing at some random location in a comment or for any arbitrary string being passed to a method or being assigned to a variable.  For something like you are asking for you'd most likely have to deeply integrate the spell checking process into each and every language which just isn't possible.  There's just not enough context available for that to happen with the classifications I do have access to either.  I could be wrong but I'd say you'd have the same issues with any spell checker that runs against source code.

Regular expression matches are an option.  I'll consider that for a future release.  The best you can do for now is build up a list of ignored words in the configuration which is effectively teaching it what not to consider a misspelling, tedious though that may be.

**Comment by cervengoc** [comment]
I understand your points, this is what I ment too when mentioned about difficoulties. Regular expressions would be a really helpful feature, thank you for considering it.

Let me have a tiny question here: where are the phrases stored when I click the "Ignore All" button in the context menu? I couln't find any configuration file which would contain them. I would like to export these ignored phrases so that I won't need to suffer the whole procedure again in the future.

**Comment by EWSoftware** [comment]
Words ignored using Ignore All are just tracked in memory for the duration of the session similar to Word or other word processors.  Once Visual Studio is restarted or if you were to open another instance, they'd be marked as misspelled again.  There isn't an easy way to add them to a configuration from the spell check context menu or  tool window since a configuration at that point may be a combination of several (global, solution, project, folder, and/or file).

I may be able to add an option such as an Always Ignore that adds it to a file much like the user dictionary though it would be global in nature rather than language specific or perhaps just always add them to the global configuration.

**Comment by cervengoc** [comment]
Hmm.. didn't even notice this yet, because I didn't close the projects since then.

It would be really good and important to have a feature like that. Maybe you could add two options, Always ignore in solution, and Always ignore globally, or so. But it would be enough only one, because then one could transfer the config file to another solution for example.

**Comment by EWSoftware** [comment]
I've opened issue #55 for ignoring spans via regex patterns.  I'm putting the option to add ignored words to a common file on hold for now.  As part of the solution/project spell check tool window implementation, I added an option to copy words from issues in the tool window's grid.  It's now a simple matter to open a configuration file of your choice, global or solution/project-specific, copy them from the grid, and paste them into the Ignored Words category configuration page to add them.  I think that achieves the same goal.
